app/controllers/trainees/diversity/disclosures_controller.rb

```diff
@@ -33,17 +33,28 @@ def disclosure_params
         params.require(:diversities_disclosure_form).permit(*Diversities::DisclosureForm::FIELDS)
       end
 
-      def redirect_to_relevant_step
-        if @disclosure_form.diversity_disclosed?
-          redirect_to(edit_trainee_diversity_ethnic_group_path(trainee))
-        else
-          trainee.apply_application? ? redirect_to(page_tracker.last_origin_page_path) : redirect_to(trainee_diversity_confirm_path(trainee))
-        end
-      end
-
       def authorize_trainee
         authorize(trainee)
       end
+
+      def step_wizard
+        @step_wizard ||= Wizards::Diversities::StepWizard.new(trainee: trainee, page_tracker: page_tracker)
+      end
+
+      def validate_form_completedness
+        return unless trainee.diversity_disclosed?
+        return if user_came_from_backlink?
+
+        redirect_to step_wizard.start_point if any_form_partially_complete?
+      end
+
+      def user_came_from_backlink?
+        session[:origin_path]&.include?("edit")
+      end
+
+      def any_form_partially_complete?
+        step_wizard.any_form_incomplete?
+      end
```

Comment:
Wondering if this can be moved into the Diversities StepWizard itself, which will reduce the public interface of the wizard to `#start_point` and `#next_step`.